### DISCUSSION TITLE
Recovery from a critical bug. Changes in content treatment in Freeplane 1.11.5_06 and 1.11.5_07

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1230] [Announcements]
Because of critical bugs introduced in Freeplane 1.11.3 which may lead to data loss and of my analysis triggered by a recent bug report, I have changed the logic of how rich text AKA html, markdown, and latex are saved. I want to release the fixed version as soon as possible. You can help by testing the latest preview 1.11.5_07. If no bugs are reported I am going to release Freeplane 1.11.5 as a stable version.

The good news: if your maps are suffering from the bug, you know it immediately because you can not load them. Then you can either recover maps from the backup. If no backup is available you might need a Freeplane script for recovering old maps. Comment here if you are affected by the bug and need me to write the script.

If you need more information about the bug you can check https://github.com/freeplane/freeplane/issues/1205

**Comment by dirkrankin** [comment]
Thanks for flagging this; I also noticed serious issues with presentation mode starting with 1.11.3, including display animation.  Could these issues be related?

**Reply by dpolivaev** [reply]
No, it is not related. Actually, I have just fixed display animation yesterday in the source code.

**Reply by dpolivaev** [reply]
Had someone reported it earlier I had probably fixed it earlier.

**Reply by dirkrankin** [reply]
Thank you!

**Comment by dpolivaev** [comment]
In the latest preview 1.11.5_16 Freeplane can read maps having the above problem and save them improving xml.

**Comment by Patronos** [comment]
Hi!

I think I found another regression in your late HTML and richtext changes that I would like to address. With other words, there are still some issues that causes further data loss within the richtext tags. Even in the *.16 release that came out yesterday or so.

Yesterday I was working on one of my MindMaps where I make use of a lot of HTML Tags. Mostly I use the 3.x specified Tags by creating Tables and keeping data stored inside it. This is an important use case of mine.

I do use <colgroup>, <col...> Tags even if I know that FreePlane doesn't use that Tag. I keep exporting the contents of the richtext nodes for further usage in other projects where I would like to have the Tables, columns and rows alinged correctly and without having the need to use excessive width tags inside the <td> elements. So the <colgroup> Tag comes in quite useful. In the past FreePlane kept ignoring these Tags and even kept them in the richtext node when saved. I sometimes use  Tags as well.

Sadly your changes removes all <colgroup> and If I saw this correctly all comment lines. I would suggest reverting these changes so the data that I entered there stays where it is.

It took my a lot of time figuring out the table sizes and the recent changes silently dropped all these meta informations that I for myself depend on.

Thank you and thanks for this cool project that I happily use on a daily basis.

**Reply by dpolivaev** [reply]
It is not a regression.
In discussion https://github.com/freeplane/freeplane/discussions/1231 the community agreed that HTML tags unknown to Freeplane should generally not be saved.
If you want us to reconsider this decision you can contribute your arguments there.

**Reply by euu2021** [reply]
I also store a lot of tables in Freeplane. I create those tables in LibreOffice Writer and then paste it in FP. Which program do you use to create the table? 

I checked some of my tables here, and, on a first sight, they seem to be working fine on FP 1.11.5-pre14.
